### PR TITLE
Reduce redundancy in .tsx files #126 and fixed a flood ping button Bug. 

### PR DESCRIPTION
**Pull request by prajwal714**
The flood ping start and stop buttons both were starting the ping. fixed the issue.
create a separate module for chart options and public Options and removed redundancy. 
@Harkishen-Singh suggest changes if any. Thanks.

### CODE REVIEW

**Comment by muskankhedia** [comment]
@prajwal714 Please have a look at the commit making rules [here](https://github.com/zairza-cetb/bench-routes#making-commits-in-bench-routes).

File: dashboard/v1.0/package-lock.json [hunk 1]
```diff
@@ -6978,11 +6978,6 @@
         }
       }
     },
-    "get-prefix": {
-      "version": "1.0.0",
-      "resolved": "https://registry.npmjs.org/get-prefix/-/get-prefix-1.0.0.tgz",
-      "integrity": "sha1-DTBUSKTjF2+cJ3F1sU4W2+b7oLU="
-    },
```
Comment (muskankhedia) [inline dashboard/v1.0/package-lock.json]: Please don't push any changes in the package-lock.json file.

File: dashboard/v1.0/package-lock.json [hunk 2]
```diff
@@ -12899,34 +12894,6 @@
...
-    "react-motion-ui-pack": {
-      "version": "0.10.3",
-      "resolved": "https://registry.npmjs.org/react-motion-ui-pack/-/react-motion-ui-pack-0.10.3.tgz",
-      "integrity": "sha512-Pb92CWNjYovx/W318D4mQPrkw/nBG1eixi7f8jV4qaHitXMh1URCgwDS9XnuJNqubv6cyVfEzMzScEFrO2/WAQ==",
-      "requires": {
-        "get-prefix": "^1.0.0",
-        "prop-types": "^15.5.10",
-        "react-motion": "^0.4.5"
-      }
-    },
```
Comment (muskankhedia) [inline dashboard/v1.0/package-lock.json]: Same here.

Comment (prajwal714) [reply dashboard/v1.0/package-lock.json]: Reverted the package-lock-json changes @muskankhedia .

**Comment by muskankhedia** [comment]
@Harkishen-Singh Please have a look for why the Travis is failing here.

**Comment by ganeshpatro321** [comment]
Hi @prajwal714  The build is failing due to unused vars in the front-end. You can refer to the attached image.

Screenshot from 2020-03-03 18-46-25

**Comment by Harkishen-Singh** [comment]
@prajwal714 please make sure that you use or fix your code by prettier before staging.

logs:

```
cd dashboard/v1.0/ && prettier '**/*.tsx' --list-different
src/components/service-ui/Ping.tsx
Makefile:14: recipe for target 'test-views' failed
make: *** [test-views] Error 1
The command "make test-views" exited with 2.
0.95s
```

File: local-config.yml [hunk 3]
```diff
@@ -51,6 +51,6 @@ utils:
     mean: 0
   services-state:
     ping: passive
-    flood-ping: passive
+    flood-ping: active
```
Comment (Harkishen-Singh) [inline local-config.yml]: `active` => `passive`

File: dashboard/v1.0/src/components/service-ui/publicOpts.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+export const opts = (operation: string, connection, type: string): void => {
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/src/components/service-ui/publicOpts.ts]: please define a type for `connection`

File: dashboard/v1.0/src/components/service-ui/publicOpts.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    switch (operation) {
+      case 'start':
+          if(type==='ping')
+          {
+             console.log("inside ping");
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/src/components/service-ui/publicOpts.ts]: remove all console statements.

**Comment by prajwal714** [comment]
@muskankhedia @Harkishen-Singh Kindly review. Suggest changes if any. Thanks.

**Comment by prajwal714** [comment]
Kindly merge the branch @Harkishen-Singh @muskankhedia 
, if the review was positive. Thanks

**Pull request merged**